easybuild/main.py

```diff
@@ -773,15 +775,19 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None, pr
                 "The following arguments will be ignored:",
             ] + orig_paths)
             print_warning(msg)
-        do_cleanup = process_easystack(options.easystack, args, logfile, testing, init_session_state, do_build)
+        is_successful = process_easystack(options.easystack, args, logfile, testing, init_session_state, do_build)
     else:
-        do_cleanup = process_eb_args(orig_paths, eb_go, cfg_settings, modtool, testing, init_session_state,
-                                     hooks, do_build)
+        is_successful = process_eb_args(orig_paths, eb_go, cfg_settings, modtool, testing, init_session_state,
+                                        hooks, do_build)
 
     # stop logging and cleanup tmp log file, unless one build failed (individual logs are located in eb_tmpdir)
     stop_logging(logfile, logtostdout=options.logtostdout)
-    if do_cleanup:
+    if is_successful:
         cleanup(logfile, eb_tmpdir, testing, silent=options.terse)
+    if options.dump_test_report or options.upload_test_report:
+        # Generation test reports is successful even when software failed to build
+        return EasyBuildExit.SUCCESS
```

Comment:
Hmm, not sure about this...

Exit code should always indicate whether (all) installations were successful or not, regardless of whether a test report was uploaded/dumped? That seems like the least surprising behavior to me...